p2p/conn.go

```diff
@@ -143,21 +149,64 @@ func (c *MultiConn) Send(topic lib.Topic, msg *Envelope) (ok bool) {
 func (c *MultiConn) startSendService() {
 	defer lib.CatchPanic(c.log)
 	m := limiter.New(0, 0)
+	ping, err := time.NewTicker(pingInterval), lib.ErrorI(nil)
+	pongTimer := time.NewTimer(pongTimeoutDuration)
 	var packet *Packet
-	defer func() { m.Done() }()
+	defer func() { lib.StopTimer(pongTimer); ping.Stop(); m.Done() }()
 	for {
 		// select statement ensures the sequential coordination of the concurrent processes
 		select {
 		case packet = <-c.streams[lib.Topic_CONSENSUS].sendQueue:
+			c.sendPacket(packet, m)
 		case packet = <-c.streams[lib.Topic_BLOCK].sendQueue:
+			c.sendPacket(packet, m)
 		case packet = <-c.streams[lib.Topic_BLOCK_REQUEST].sendQueue:
+			c.sendPacket(packet, m)
 		case packet = <-c.streams[lib.Topic_TX].sendQueue:
+			c.sendPacket(packet, m)
 		case packet = <-c.streams[lib.Topic_PEERS_RESPONSE].sendQueue:
+			c.sendPacket(packet, m)
 		case packet = <-c.streams[lib.Topic_PEERS_REQUEST].sendQueue:
+			c.sendPacket(packet, m)
```

Comment:
although this whole logic makes sense to be here as is dealing with sending (and receiving on `startReceiveService`, I think this feels odd that all the types of messages follow the same pattern and ping/pong breaks it completely, how about doing a `startPingPongService` and move the logic for both send and receive there, could call two goroutines there to handle each select and have the timer logic outside as is happening here, do let me know what you think

Reply:
I like more for them to be on the already existing send and receive channels so we don't add more channels.

Reply:
I agree with @pablocampogo I think this is fine